vercel/turbo
# fix(turborepo): non-determinism with root internal deps hash.

### Description

We were getting different task hashes with each run in some monorepos with root internal dependencies. Turns out, the file hashes change depending on which order the packages are hashed. No clue why, but to solve this, we sort the packages before hashing.

### Testing Instructions

Try running manually and check that hashes stay the same. Also added integration test confirming root internal deps behavior
